Sources/MuxPlayerSwift/FairPlay/ContentKeySessionDelegate.swift

```diff
@@ -216,19 +238,80 @@ class ContentKeySessionDelegate<SessionManager: FairPlayStreamingSessionManager>
         group.wait()
         
         guard let ckcData = ckcData else {
-            print("no CKC Data in CKC response")
             request.processContentKeyResponseError(FairPlaySessionError.unexpected(message: "No CKC Data returned from CDM"))
             return
         }
         
         print("Submitting CKC to system")
         // Send CKC to CDM/wherever else so we can finally play our content
-        let keyResponse = AVContentKeyResponse(fairPlayStreamingKeyResponseData: ckcData)
+//        let keyResponse = AVContentKeyResponse(fairPlayStreamingKeyResponseData: ckcData)
+        let keyResponse = request.makeContentKeyResponse(data: ckcData)
         request.processContentKeyResponse(keyResponse)
         // Done! no further interaction is required from us to play.
     }
 }
 
+// Wraps a generic request for a key and delegates calls to it
+//  this protocol's methods are intended to match AVContentKeyRequest
+protocol KeyRequest {
+    
+    associatedtype InnerRequest
+    
+    var identifier: Any? { get }
+    
+    func makeContentKeyResponse(data: Data) -> AVContentKeyResponse
+    
+    func processContentKeyResponse(_ response: AVContentKeyResponse)
+    func processContentKeyResponseError(_ error: any Error)
+    func makeStreamingContentKeyRequestData(forApp appIdentifier: Data,
+                                            contentIdentifier: Data?,
+                                            options: [String : Any]?,
+                                            completionHandler handler: @escaping (Data?, (any Error)?) -> Void)
+}
+
+// Wraps a real AVContentKeyRequest and straightforwardly delegates to it
+struct DefaultKeyRequest : KeyRequest {
```

Comment:
Yup

Reply:
Pretty much yes.

If we ever supported clearkey (or something else) this might aid re-usability though, since it has a factory for `AVContentKeyResponse`, which must be constructed differently for different schemes.